core/src/main/java/org/infinispan/metrics/impl/CacheManagerMetricsRegistration.java

```diff
@@ -13,7 +13,17 @@
 public final class CacheManagerMetricsRegistration extends AbstractMetricsRegistration {
 
    @Override
-   protected String initPrefix() {
+   protected String initNamePrefix() {
       return "CacheManager_" + globalConfig.cacheManagerName();
    }
+
+   @Override
+   protected String initNodeName() {
+      String nodeName = globalConfig.transport().nodeName();
+      if (nodeName == null || nodeName.isEmpty()) {
+         //TODO [anistor] ensure unique node name is set in all tests and also in real life usage
+         nodeName = java.util.UUID.randomUUID().toString();
```

Comment:
Replace with Util.threadLocalRandomUUID()

Reply:
This is executed once, on startup. Why bother with optimizations?

Reply:
We want to do it in because of https://issues.redhat.com/browse/ISPN-11230

Reply:
So "it computes an expensive cryptographic digest". How expensive? Even if it's expensive it cannot cost a ton. Why do we care if it's done only once per cache manager? Can @danberindei elaborate a bit ?

Reply:
Ignore that for now.

Reply:
It's not too expensive to do once per cache manager, but IMO it's good to have a simple rule of using `Util.threadLocalRandomUUID()` everywhere instead of having long discussions about whether it's worth optimizing a particular call or not every single time.

Reply:
I forgot to mention that I would have preferred to set the node name to hostname + random number when the user hasn't specified it, the way JGroups does it.

Reply:
Then I would gladly grab that generated string from JGroups rather than generating a new meaningless uuid myself. The intention was to have a meaningful tag per metric that traces back to the node that generated that data. Regarding the long discussion about perf implications of this vs that, they tend to get long only when they miss the point. This is not perf relevant code and we should not advocate use of our 'patched' versions of various jdk utils we have in our commons. Those are accidents and should not turn into irrelevant rules. -1.02 for that checkstyle rule or whatver gatekeeper you plan to set up.

Reply:
If you can wait for `JGroupsTransport` to start and connect the channel, then yes, please do that. A random UUID won't trace back to the node that generated it, regardless of how those random bits are generated.